printer.cfg

```diff
@@ -35,6 +35,7 @@
 # [include config/hardware/base/fans/controller_fan.cfg]
 # [include config/hardware/base/fans/exhaust_fan.cfg]
 
+[include config/hardware/base/displays/mini12864.cfg]
```

Comment:
Can you add a `#` before this line to avoid having it included by default ? Thanks.

Also if possible to put the display files in the `hardware/addons` folder instead as it's not mandatory to have one to run a machine (I don't have any on every of my machines).

And finally, also please add the line corresponding to the minidisplay lights and caselights like following:
```
# [include config/hardware/addons/lights/neopixel_mini12864.cfg]
# [include config/hardware/addons/lights/neopixel_caselight.cfg]
```

Reply:
Ok but not agree to put the mini12864.cfg file in the addon structure. That item explicitely integrated into the official bom and is delivered for all kits now. Of course not mandatory to work but for my point of view'd remain in base directory structure.

Reply:
@zebons but it isn't a necessary hardware. I think addon is quite right. I would suggest to add a display section:
config/hardware/addons/display/neopixel_mini12864.cfg would be suting since there are a lot of different displays.

The LED settings can be added to the lights section though.

Reply:
In that case, we can also say that the exhaust fan isn't mandatory for printing, so it'd be put in the addon directory....

And the fact that there are a lot of differents displays isn't a decisive point to put it in the addon directory. There are also a lot of mcus but they must stay in the base structure.

To simplify we'd take into account the official bom and documentation and all elements described here'd be in the base structure and all componants not notified here in the addon one.

But if you want to deplace it, you can do.

Reply:
Hello :)

So, I read your opinions on the topic and your points of view are interesting.
Here is mine: the structure of this repo is not designed to follow the Voron BOM as I want it to be usable on all the printers having the same kind of mechanics like VzBot, etc... FYI, I also personnally use it on a Prusa and I know people also using it on multiple Klipperized Ender3.

So in my opinion, the difference between base vs addon in this config is more on what is needed to run a printer under Klipper vs what is not mandatory. That's why I prefer to put the displays in the `addons/displays` folder.

Currently the exauhst fan and controller fan are placed in the `base/fans` folder as I put all the fans here without really thinking about it. Now, you made me rethink about it and I agree that it could be better to put the controller_fan in a new `addons/fans` folder. Regarding the exhaust fan, I think that we can just move it in the `addons/filters` folder.

Reply:
Hi,
I was indeed thinking mainly of Vorons and not of portability to other printers, so I understand your position better. And it becomes clearer if each component is more clearly put into the right structure which I didn't think was the case. 
So the exhaust fan should be placed in the addon structure. 

But this reorganization leads to two remarks. 
The first is that it remains your interpretation of what is mandatory or not and will not necessarily be for others as this discussion proves. 
The second is that I thought it would be simpler and more logical to group all the fans, for example, in the same directory. And this will lead to even more misunderstandings and questions...

In the end, and given the number of files, I wonder if it wouldn't have been much simpler not to make this base addon distinction and simply classify the components by type. I often wander through the tree to find my files because of this division and it's a bit of a shame 

Translated with www.DeepL.com/Translator (free version)

Reply:
@zebons Agreed, I also find myself a little bit lost in the config files. I would also suggest to have the "to be configured by the user" files seperated from the "don't touch me" files

Reply:
Mhmm yeah ok for me, it make sense then.

So I'll do a final review now and merge it as it is today. Then I'll take some time to refactor correctly by removing the base vs addons folders and simplify the folder structure.

Thank you for your insights !

Reply:
Mhmm yeah ok for me, it make sense then.

So I'll do a final review now and merge it as it is today. Then I'll take some time to refactor correctly by removing the base vs addons folders and simplify the folder structure.

Thank you for your insights !